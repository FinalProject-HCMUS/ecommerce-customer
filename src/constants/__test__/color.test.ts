import colorData from '../color';

describe('colorData', () => {
  test('should export an object', () => {
    expect(colorData).toBeDefined();
    expect(typeof colorData).toBe('object');
  });

  test('should contain all expected color keys', () => {
    const expectedColors = [
      'purple',
      'yellow',
      'black',
      'white',
      'pink',
      'gray',
      'teal',
      'maroon',
      'navy',
      'turquoise',
      'lavender',
      'peach',
      'silver',
      'gold',
      'burgundy',
      'olive',
      'mustard',
      'mint',
      'coral',
      'taupe',
    ];

    expectedColors.forEach((color) => {
      expect(colorData).toHaveProperty(color);
    });

    // Verify we're not missing any colors or have extra ones
    expect(Object.keys(colorData).sort()).toEqual(expectedColors.sort());
  });

  test('should have valid hex color codes for all values', () => {
    const hexColorRegex = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;

    Object.values(colorData).forEach((colorCode) => {
      expect(colorCode).toMatch(hexColorRegex);
    });
  });

  test('should have correct hex values for specific colors', () => {
    expect(colorData.purple).toBe('#800080');
    expect(colorData.yellow).toBe('#FFFF00');
    expect(colorData.black).toBe('#000000');
    expect(colorData.white).toBe('#FFFFFF');
    expect(colorData.pink).toBe('#FFC0CB');
    expect(colorData.turquoise).toBe('#40E0D0');
    expect(colorData.gold).toBe('#FFD700');
  });

  test('should match snapshot', () => {
    expect(colorData).toMatchSnapshot();
  });
});
