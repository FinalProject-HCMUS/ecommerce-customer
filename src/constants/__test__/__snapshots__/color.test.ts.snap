// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorData should match snapshot 1`] = `
{
  "black": "#000000",
  "burgundy": "#800020",
  "coral": "#FF7F50",
  "gold": "#FFD700",
  "gray": "#808080",
  "lavender": "#E6E6FA",
  "maroon": "#800000",
  "mint": "#98FF98",
  "mustard": "#FFDB58",
  "navy": "#000080",
  "olive": "#808000",
  "peach": "#FFDAB9",
  "pink": "#FFC0CB",
  "purple": "#800080",
  "silver": "#C0C0C0",
  "taupe": "#483C32",
  "teal": "#008080",
  "turquoise": "#40E0D0",
  "white": "#FFFFFF",
  "yellow": "#FFFF00",
}
`;
